ampd/src/commands/daemon.rs

```diff
@@ -0,0 +1,26 @@
+use error_stack::{Report, ResultExt};
+use std::path::PathBuf;
+
+use crate::config::Config;
+use crate::state::{flush, load};
+use crate::Error;
+
+pub async fn run(config: Config, state_path: PathBuf) -> Result<(), Report<Error>> {
```

Comment:
use either `AsRef<Path>` or `&Path`, `PathBuf` is only necessary to correctly parse the starting args. It's similar to `String` and `&str`, so `PathBuf` makes a full copy of the content